internal/filesystem/nfs/client.go

```diff
@@ -0,0 +1,611 @@
+package nfs
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path"
+	"strings"
+	"sync"
+
+	"github.com/djdv/go-filesystem-utils/internal/filesystem"
+	fserrors "github.com/djdv/go-filesystem-utils/internal/filesystem/errors"
+	"github.com/djdv/go-filesystem-utils/internal/generic"
+	"github.com/multiformats/go-multiaddr"
+	manet "github.com/multiformats/go-multiaddr/net"
+	"github.com/willscott/go-nfs-client/nfs"
+	"github.com/willscott/go-nfs-client/nfs/rpc"
+)
+
+type (
+	goFS struct {
+		target        *nfs.Target
+		linkSeparator string
+		linkLimit     uint
+		// NOTE [2024.01.08]: The NFS server library is able to handle multiple requests concurrently
```

Comment:
the lock is unfortunate - there may be a better nfs client out there that doesn't have the concurrency limitation.

Reply:
For sure.
I'd be open to suggestions on alternatives.
But it might also be worth filing an issue about it and hoping it gets resolved eventually.
Can't commit to looking into it myself right now, but it could happen eventually. Of course anyone else could pop in for it too.

Reply:
FWIW it looks like the issue tracker for [your fork](https://github.com/willscott/go-nfs-client) is disabled.
If that's intentional, no problem.